### PR TITLE
golioth: coap_utils: new helper module

### PR DESCRIPTION
**Pull request by mniestroj**
This private module will be used by updated lightdb and fw components.

### CODE REVIEW

**Comment by mniestroj** [comment]
I believe that checkpatch warning is a false-positive:
```
389a471d9d3d9ca74371d338c1419ba976c4e381:41: CHECK:MACRO_ARG_PRECEDENCE: Macro argument '_name' may be better as '(_name)' to avoid precedence issues
#41: FILE: net/golioth/pathv.h:6:
+#define PATHV(_name, ...)				\
+	const uint8_t *_name[] = {			\
+		FOR_EACH(IDENTITY, (,), __VA_ARGS__),	\
+		NULL, /* sentinel */			\
+	}
```

**Review by ncmiller — APPROVED** [review]
Nice work, just one minor comment.

File: net/golioth/pathv.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#ifndef __NET_GOLIOTH_PATHV_H__
+#define __NET_GOLIOTH_PATHV_H__
+
+#include <zephyr/sys/util.h>
+
+#define PATHV(_name, ...)				\
```
[the remark this answers is not in the document]
Comment (mniestroj) [reply net/golioth/pathv.h]: I was going back and forth with that one before creating PR. Decided to keep `PATHV()` as long as it is private. Though, I don't have strong opinion on: `GOLIOTH_PATHV` vs `PATHV`.

**Comment by mniestroj** [comment]
@ncmiller Please take another look at the `PATHV()` definition. I decided to use "anonymous compound literals" as I have run into such code in Zephyr CTF (common tracing format) implementation. Seems to be a standard starting with C99, so safe to use here. This allows to use `PATHV()` directly in function invocation as opposed to declaring temporary variable explicitly in function body.

**Review by ncmiller — APPROVED** [review]

**Pull request merged**